File: server/src/main/java/com/exam/examMS/service/QuizService.java
package com.exam.examMS.service;

import java.util.Set;

import com.exam.examMS.entity.exam.Category;
import com.exam.examMS.entity.exam.Quiz;

public interface QuizService {


	public Quiz addQuiz(Quiz quiz);
	public Quiz updateQuiz(Quiz quiz);
	public Set<Quiz> getQuiz();
	public Quiz getQuiz(Long quizId);
	public Set<Quiz> getQuizByCategory(Category category);
	public Set<Quiz> getActiveQuizzes();
	public Set<Quiz> getActiveQuizzesOfCategory(Category category);
	public void deleteQuiz(Long quizId);
}

File: client/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private _http:HttpClient) { }

  public getQuizzes()
  {
    return this._http.get(`${baseUrl}/quiz/`);  
  }

  public addQuizzes(quiz:any)
  {
      return this._http.post(`${baseUrl}/quiz/`,quiz);
  }

  public deleteQuiz(qId:any)
  {
      return this._http.delete(`${baseUrl}/quiz/${qId}`);
  }

  public updateQuiz(quiz:any)
  {
      return this._http.put(`${baseUrl}/quiz/`,quiz);
  }

  public getQuiz(qId:any)
  {
    return this._http.get(`${baseUrl}/quiz/${qId}`); 
  }

  public getActiveQuizzes()
  {
    return this._http.get(`${baseUrl}/quiz/active`); 
  }

  public getQuizByCategory(catId:any)
  {
    return this._http.get(`${baseUrl}/quiz/category/${catId}`); 
  }

  public getQuizByCategoryActive(catId:any)
  {
    return this._http.get(`${baseUrl}/quiz/category/active/${catId}`); 
  }
}
